### PR TITLE
feat(mapping): introduce osarchmap per issue #142

### PR DESCRIPTION
**Pull request by noelmcloughlin**
As proposed and discussed in #142

### CODE REVIEW

**Comment by myii** [comment]
@noelmcloughlin Do you mean #142 instead of #13?  Otherwise, I've got to come back to this to run a few checks but thanks for contributing this PR.

**Comment by noelmcloughlin** [comment]
Good catch - my bad - thanks @myii

**Comment by myii** [comment]
@noelmcloughlin Tested this with `salt-ssh` on a couple of minions and all looking good, thanks.  I'm just wondering whether it's worth expanding this a little to add some tests for this.  Even if only testing to confirm that `arch: amd64` is coming through in the map.  Or perhaps even adding another configuration option and testing for that.  What do you think?

File: test/integration/default/controls/config_spec.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@
     its('content') { should include '"pkg": {"name": "' }
     its('content') { should include '"service": {"name": "' }
     its('content') { should include '"tofs": {"files_switch": ["any/path/can/be/used/here", "id", "osfinger", "os", "os_family"], "source_files": {"template-config-file-file-managed": ["example.tmpl.jinja"]}' }
+    its('content') { should include '"arch": "amd64"}' }
```
Comment (myii) [inline test/integration/default/controls/config_spec.rb]: This works:

```suggestion
    its('content') { should include '"arch": "amd64"' }
```

Comment (noelmcloughlin) [reply test/integration/default/controls/config_spec.rb]: Thats fixed now. yes go ahead and make commits!! thanks @myii

**Comment by noelmcloughlin** [comment]
Hi @myii is this good enough yet?

**Comment by myii** [comment]
@noelmcloughlin Do you mind if I add an extra commit to this PR, just for the comments I mentioned earlier?  Also, the main branch of this repo has been changed to `develop` (not `master`) after this PR was started.  That can be done from this end, though.  So I can do both if that's OK with you.

**Comment by noelmcloughlin** [comment]
Sure!!!

**Comment by myii** [comment]
There we go, @noelmcloughlin -- merged.  Sorry, I didn't notice you'd already said it was OK to add the commit (https://github.com/saltstack-formulas/template-formula/pull/144#discussion_r298811332).

**Pull request merged**